packages/diagrams/lib/src/text.dart

```diff
@@ -484,6 +486,514 @@ class TextHeightComparisonPainter extends CustomPainter {
+      );
+    }
+
+    if (glyphsBox.bottom >= size.height - 1)
+      return;
+    // Bottom Leading Area
+    labelPainter.text = TextSpan(text: 'Bottom Leading', style: textStyle);
+    labelPainter.layout();
+    final Rect bottomLeadingArea = Offset(0, glyphsBox.bottom) & Size(size.width, size.height - glyphsBox.bottom);
+    final Rect bottomTextRect = Rect.fromCenter(
+      center: bottomLeadingArea.center,
+      width: labelPainter.width,
+      height: labelPainter.height,
+    );
+    labelPainter.paint(canvas, bottomTextRect.topLeft);
+
+    drawPointedLine(
+      -fontMetricsLabelPadding + bottomTextRect.topCenter,
+      fontMetricsLabelPadding + bottomLeadingArea.topCenter,
+    );
+
+    drawPointedLine(
+      fontMetricsLabelPadding + bottomTextRect.bottomCenter,
+      fontMetricsLabelPadding * -2 + bottomLeadingArea.bottomCenter,
+    );
+  }
+
+  void _paintHeightIndicator(Canvas canvas, Size size) {
+    final Paint paint = Paint();
+    // Height bracket & label
+    paint.strokeWidth = 2;
+    paint.style = PaintingStyle.stroke;
+    paint.color = Colors.blue[900]!;
+
+    final Path path = Path();
+    path.moveTo(- 10, 0);
+    path.lineTo(heightCaptionBracketMinX, 0);
+    path.lineTo(heightCaptionBracketMinX, size.height);
+    path.lineTo(- 10, size.height);
+    canvas.drawPath(path, paint);
+
+    // Layout is done in performLayout.
+    final Offset origin = Offset(
+      heightCaptionBracketMinX - heightCaptionTextPainter.width,
+      (size.height - heightCaptionTextPainter.height) / 2,
+    ) - fontMetricsLabelPadding;
+    heightCaptionTextPainter.paint(canvas, origin);
+  }
+
+  void _paintCaptions(Canvas canvas, Size size, Rect glyphsBox, double baseline) {
+    // Ascent
+    final TextPainter labelPainter = TextPainter(
+      text: TextSpan(
+        text:'Font Ascent',
+        style: TextStyle(
+          fontSize: 9,
+          color: Colors.red[900]!,
+        ),
+      ),
+      textAlign: TextAlign.right,
+      textDirection: textDirection,
+    );
+    labelPainter.layout();
+    Offset labelPaintOrigin = Offset(0, glyphsBox.top - labelPainter.height) - fontMetricsLabelPadding;
+    labelPainter.paint(canvas, labelPaintOrigin);
+
+    // Descent
+    labelPainter.text = TextSpan(
+      text: 'Font Descent',
+      style: TextStyle(
+        fontSize: 9,
+        color: Colors.red[900]!,
+      ),
+    );
+    labelPainter.layout();
+    labelPaintOrigin = Offset(0, glyphsBox.bottom) + fontMetricsLabelPadding;
+    labelPainter.paint(canvas, labelPaintOrigin);
+
+     // Baseline
+    labelPainter.text = const TextSpan(
+      text: 'Alphabetic Baseline',
+      style: TextStyle(
+        fontSize: 9,
+        color: Colors.black,
+      ),
+    );
+    labelPainter.layout();
+    labelPaintOrigin = Offset(0, baseline) + fontMetricsLabelPadding;
+    labelPainter.paint(canvas, labelPaintOrigin);
+
+    // Top
+    labelPainter.text = TextSpan(
+      text: 'Line-Box Top',
```

Comment:
Not sure this is the best term. "line box" in css seems to have a different meaning:  https://www.w3.org/TR/CSS2/visuren.html#inline-formatting

> The rectangular area that contains the boxes that form a line is called a line box.

Reply:
Yeah it seems very similar but not quite.  As I understand it, the CSS definition would be the bounding height of all inline elements in the line, but as you've defined it, it's the local height of the current span.

Related to my comment above about hyphens, what would this be in Flutter?  Is it the same as the "intrinsic height" of the render box?